source/geod24/RingBuffer.d

```diff
@@ -0,0 +1,127 @@
+module geod24.RingBuffer;
+
+import std.range;
+
+/*******************************************************************************
+
+    A ring buffer implementation with fixed size
+
+    Params:
+        T = Type of elements
+
+*******************************************************************************/
+
+class RingBuffer (T)
+{
+
+    /*******************************************************************************
+
+        Params:
+            size = Maximum number of elements
+
+    *******************************************************************************/
+
+    this (size_t size) @safe pure nothrow
+    {
+        this.storage = new T[size];
+        this.storage.length = size;
```

Comment:
This is is sufficient:

```suggestion
        this.storage.length = size;
```

Slices don't need to be `new`-ed, although they can. But either one of these lines you wrote would do the exact same thing.